src/syscall.c

```diff
@@ -327,13 +328,11 @@ static void syscall_open(riscv_t *rv)
     uint32_t mode = rv_get_reg(rv, rv_reg_a2);
 
     /* read name from runtime memory */
-    char name_str[256] = {'\0'};
-    uint32_t read = memory_read_str(attr->mem, (uint8_t *) name_str, name,
-                                    sizeof(name_str));
-    if (read > sizeof(name_str)) {
-        rv_set_reg(rv, rv_reg_a0, -1);
-        return;
-    }
+    const size_t name_len = strlen(attr->mem->mem_base + name);
+    char *name_str = malloc(name_len + 1);
```

Comment:
Would the allocated memory be returned back later?

Reply:
> Would the allocated memory be returned back later?

I should free it after `fopen`.